io.github.pastthepixels.InfiniteShooter.yml

```diff
@@ -0,0 +1,32 @@
+app-id: io.github.pastthepixels.InfiniteShooter
+runtime: org.freedesktop.Platform
+runtime-version: 21.08
+sdk: org.freedesktop.Sdk
+command: InfiniteShooter.x86_64
+finish-args:
+  - --share=ipc
+  - --socket=x11 # <-- Godot needs X11
+  - --socket=pulseaudio
+  - --device=all # <-- This flag enables controller support
+modules:
+  # InfiniteShooter (the game)
+  - name: InfiniteShooter
+    buildsystem: simple
+    build-commands:
+      - install -D InfiniteShooter-LINUX.x86_64 /app/bin/InfiniteShooter.x86_64
+    sources:
+      - type: file
+        url: https://github.com/pastthepixels/InfiniteShooter/releases/download/v0.4.0/InfiniteShooter-LINUX.x86_64 # <-- Edit with latest verion
+        sha256: 4c483a24895adaa4dc46500b2102b18eb31bc3adbe685cd0e7e55c38d7554386                                    # <--
+
+  # InfiniteShooter's .desktop shortcut + icon
+  - name: InfiniteShooter-Shortcut
+    buildsystem: simple
+    build-commands:
+      - install -D io.github.pastthepixels.InfiniteShooter.desktop /app/share/applications/io.github.pastthepixels.InfiniteShooter.desktop # more like install XD
+      - install -D io.github.pastthepixels.InfiniteShooter.png /app/share/icons/hicolor/256x256/apps/io.github.pastthepixels.InfiniteShooter.png
+      - install -D io.github.pastthepixels.InfiniteShooter.appdata.xml /app/share/metainfo/io.github.pastthepixels.InfiniteShooter.appdata.xml
+    sources:
+      - type: archive
+        url: https://github.com/pastthepixels/InfiniteShooter-Flatpak/archive/refs/tags/v0.4.0.zip # <-- Edit with latest version
```

Comment:
you could as well have one tarball since a desktop app shipped without a .desktop or an icon is a bug...

Reply:
Thanks for the suggestion! I'm not sure if I'm really allowed to do this, but I think I've seen some other programs do this, so I'm thinking I'll move the desktop file, .appdata.xml, and icon over to this branch with the manifest and just pull the game data from the cloud. Then I can move everything from InfniteShooter-Shortcut to one module and work from there.
(Also, I should really apologize if I sound like I don't know what I'm doing or I'm making things take longer than they need to. Sorry about that!)

Reply:
It's OK, we're here to help and most of the things that are suggested or requirements are, as hub says, there to try and make your app a *better Linux app* not just for flatpak (although it should make it easier to maintain for you as well in the medium term we think)